ftw/linkchecker/command/checking_links.py

```diff
@@ -7,28 +7,47 @@
 from zope.component import getUtility
 
 
-def main(plone, *args):
-
-    # Set up request for debug / bin/instance run mode.
-    app = makerequest(plone)
+def get_plone_sites_information(app):
+    # make fake request to be able to traverse
+    app = makerequest(app)
     setRequest(app.REQUEST)
+    overview_view = app.unrestrictedTraverse('plone-overview')
+    plone_sites = overview_view.sites()
+    # create information dictionary for each plone site
+    plone_sites_information = [{
+        'id': plone_site.getId(),
+        'path': '/'.join(plone_site.getPhysicalPath()),
+        'title': plone_site.Title()
+    } for plone_site in plone_sites]
 
-    houptding = app.restrictedTraverse('/')
+    return plone_sites_information
 
-    # here all pages need to be found
-    # for now there is only 'Plone'
-    # later on we need to iterate through
-    # each page and send reports to different people
 
-    MINI_SITE = houptding.get('Plone')
+def get_relation_catalog_for_plone_site(app, PloneSiteId):
+    EINI_VODE_SITENE = app.get(PloneSiteId)
 
     user = AccessControl.SecurityManagement.SpecialUsers.system
-    user = user.__of__(MINI_SITE.acl_users)
-    newSecurityManager(MINI_SITE, user)
+    user = user.__of__(EINI_VODE_SITENE.acl_users)
+    newSecurityManager(EINI_VODE_SITENE, user)
+
+    setSite(EINI_VODE_SITENE)
+
+    return getUtility(ICatalog)
+
+
+def suech_sache_im_katalog(relation_catalog):
```

Comment:
WTF? https://devdocs.4teamwork.ch/programmierrichtlinien/#einhalten-der-pep8-konventionen

Reply:
Thanks a lot for your review. I think I'm closer to the conventions now.

I was rather thinking of a review concerning the content and hope to get some advise on that too ;)